cns/middlewares/k8sSwiftV2_linux.go

```diff
@@ -0,0 +1,97 @@
+package middlewares
+
+import (
+	"fmt"
+	"net/netip"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/Azure/azure-container-networking/cns/configuration"
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/middlewares/utils"
+	"github.com/pkg/errors"
+)
+
+// setRoutes sets the routes for podIPInfo used in SWIFT V2 scenario.
+func (k *K8sSWIFTv2Middleware) setRoutes(podIPInfo *cns.PodIpInfo) error {
+	logger.Printf("[SWIFTv2Middleware] set routes for pod with nic type : %s", podIPInfo.NICType)
+	var routes []cns.Route
+
+	switch podIPInfo.NICType {
+	case cns.DelegatedVMNIC:
+		virtualGWRoute := cns.Route{
+			IPAddress: fmt.Sprintf("%s/%d", virtualGW, prefixLength),
+		}
+		// default route via SWIFT v2 interface
+		route := cns.Route{
+			IPAddress:        "0.0.0.0/0",
+			GatewayIPAddress: virtualGW,
+		}
+		routes = append(routes, virtualGWRoute, route)
+
+	case cns.InfraNIC:
+		// Get and parse infraVNETCIDRs from env
+		infraVNETCIDRs, err := configuration.InfraVNETCIDRs()
+		if err != nil {
+			return errors.Wrapf(err, "failed to get infraVNETCIDRs from env")
+		}
+		infraVNETCIDRsv4, infraVNETCIDRsv6, err := utils.ParseCIDRs(infraVNETCIDRs)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse infraVNETCIDRs")
+		}
+
+		// Get and parse podCIDRs from env
+		podCIDRs, err := configuration.PodCIDRs()
+		if err != nil {
+			return errors.Wrapf(err, "failed to get podCIDRs from env")
+		}
+		podCIDRsV4, podCIDRv6, err := utils.ParseCIDRs(podCIDRs)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse podCIDRs")
+		}
+
+		// Get and parse serviceCIDRs from env
+		serviceCIDRs, err := configuration.ServiceCIDRs()
+		if err != nil {
+			return errors.Wrapf(err, "failed to get serviceCIDRs from env")
+		}
+		serviceCIDRsV4, serviceCIDRsV6, err := utils.ParseCIDRs(serviceCIDRs)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse serviceCIDRs")
+		}
+
+		ip, err := netip.ParseAddr(podIPInfo.PodIPConfig.IPAddress)
+		if err != nil {
+			return errors.Wrapf(err, "failed to parse podIPConfig IP address %s", podIPInfo.PodIPConfig.IPAddress)
+		}
+
+		if ip.Is4() {
+			routes = append(routes, addRoutes(podCIDRsV4, overlayGatewayv4)...)
+			routes = append(routes, addRoutes(serviceCIDRsV4, overlayGatewayv4)...)
+			routes = append(routes, addRoutes(infraVNETCIDRsv4, overlayGatewayv4)...)
+		} else {
+			routes = append(routes, addRoutes(podCIDRv6, overlayGatewayV6)...)
+			routes = append(routes, addRoutes(serviceCIDRsV6, overlayGatewayV6)...)
+			routes = append(routes, addRoutes(infraVNETCIDRsv6, overlayGatewayV6)...)
+		}
+		podIPInfo.SkipDefaultRoutes = true
+
+	case cns.BackendNIC, cns.NodeNetworkInterfaceBackendNIC, cns.NodeNetworkInterfaceAccelnetFrontendNIC:
```

Comment:
let's remove the first nic-type case here. it is not in use for linux today & we will be deprecating it so we can remove it.

Reply:
Removed.